processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -1135,21 +1117,21 @@ public MetricDesc(int index, AggregatorFactory factory)
       this.index = index;
       this.name = factory.getName();
 
-      String typeInfo = factory.getTypeName();
-      if ("float".equalsIgnoreCase(typeInfo)) {
-        capabilities = ColumnCapabilitiesImpl.createSimpleNumericColumnCapabilities(ValueType.FLOAT);
-        this.type = typeInfo;
-      } else if ("long".equalsIgnoreCase(typeInfo)) {
-        capabilities = ColumnCapabilitiesImpl.createSimpleNumericColumnCapabilities(ValueType.LONG);
-        this.type = typeInfo;
-      } else if ("double".equalsIgnoreCase(typeInfo)) {
-        capabilities = ColumnCapabilitiesImpl.createSimpleNumericColumnCapabilities(ValueType.DOUBLE);
-        this.type = typeInfo;
+      ValueType valueType = factory.getType();
+
+      if (valueType.isPrimitive()) {
```

Comment:
Previously this only did createSimpleNumericColumnCapabilities if the type was numeric, now it includes strings too. Is that intentional?

Reply:
Hmm, that is a good point, this should probably check `isNumeric`. However, should the `else` be an `else if (ValueType.COMPLEX.equals(valueType))` to make sure it is actually is? I'm not sure the previous behavior of treating `STRING` as a `COMPLEX` was quite correct, it doesn't seem like it. Instead it should maybe handle `STRING` (and arrays) separately, though I'm not quite sure which capabilities it should have or if it should be an error case.

Reply:
Maybe throw an error that this kind of AggregatorFactory isn't supported at ingestion time?

I don't think there's a use case for it with any of the builtin aggregators. So we can worry about it when one comes up.

Reply:
Changed to handle only numeric and complex types, and left a comment about what needs done if the new `else` case is encountered.